membership-attribute-service/app/actions/CommonActions.scala

```diff
@@ -22,24 +22,30 @@ class CommonActions(touchpointBackends: TouchpointBackends, bodyParser: BodyPars
   def noCache(result: Result): Result = NoCache(result)
 
   val NoCacheAction = resultModifier(noCache)
-  def AuthAndBackendViaAuthLibAction(requiredScopes: List[AccessScope]) =
+  def AuthAndBackendViaAuthLibAction(requiredScopes: List[AccessScope]): ActionBuilder[AuthenticatedUserAndBackendRequest, AnyContent] =
     NoCacheAction andThen new AuthAndBackendViaAuthLibAction(touchpointBackends, requiredScopes, isTestUser)
-  def AuthAndBackendViaIdapiAction(howToHandleRecencyOfSignedIn: HowToHandleRecencyOfSignedIn) =
+  def AuthAndBackendViaIdapiAction(howToHandleRecencyOfSignedIn: HowToHandleRecencyOfSignedIn): ActionBuilder[AuthAndBackendRequest, AnyContent] =
     NoCacheAction andThen new AuthAndBackendViaIdapiAction(touchpointBackends, howToHandleRecencyOfSignedIn, isTestUser)
 
+  def AuthAndBackendViaBothAction(
+      howToHandleRecencyOfSignedIn: HowToHandleRecencyOfSignedIn,
+      requiredScopes: List[AccessScope],
+  ): ActionBuilder[AuthenticatedUserAndBackendRequest, AnyContent] =
+    NoCacheAction andThen
+      new AuthAndBackendViaIdapiAction(touchpointBackends, howToHandleRecencyOfSignedIn, isTestUser) andThen
+      new AuthAndBackendViaAuthLibAction(touchpointBackends, requiredScopes, isTestUser)
```

Comment:
Not sure why this new action is necessary

Reply:
What's the alternative when we need to check for both recency of login and scopes?

Reply:
Scopes only apply when authorising with Okta so initially could just use `AuthAndBackendViaIdapiAction` for Idapi auth.  Okta support is coming to that action imminently, certainly within the next week.